### PR TITLE
Fix test case backlog.population

### PR DESCRIPTION
**Pull request by dsiganos**
The problem is that node destruction is not managed and any callback we register must not assume that the resources it accesses are alive

### CODE REVIEW

**Comment by pwojcikdev** [comment]
Don't do it like that, this makes code meesy. Moving mutex & set before system ensures the correct destruction order.

File: nano/core_test/backlog.cpp [hunk 1]
```diff
@@ -75,32 +75,44 @@ TEST (backlog, population)
...
 	auto & node = *system.add_node ();
 
-	nano::mutex mutex;
-	std::unordered_set<nano::account> activated;
-
-	node.backlog.activate_callback.add ([&] (nano::transaction const & transaction, nano::account const & account, nano::account_info const & account_info, nano::confirmation_height_info const & conf_info) {
-		nano::lock_guard<nano::mutex> lock{ mutex };
-
-		activated.insert (account);
+	// the node destruction happens in the background and parts of the node object
```
Comment (pwojcikdev) [inline nano/core_test/backlog.cpp:78]: Well, this comment is not true. `system` destructor ensures node will be stopped and therefore all relevant threads will also be stopped.

Comment (dsiganos) [reply nano/core_test/backlog.cpp:78]: I am certain that parts of the node survive even after system destruction. I have spent a month last year due to that problem and in the end I resorted to this pattern of using weak pointers in callbacks to make them safe.

**Comment by dsiganos** [comment]
@pwojcikdev if you can stop it failing just by moving the mutex before the system initialisation, then be guest, you do it.
I know for certain that that is not a fix.

File: nano/core_test/backlog.cpp (hunk 1 above)
Comment (pwojcikdev) [reply nano/core_test/backlog.cpp:78]: You are confusing IO threads with component threads.

Comment (pwojcikdev) [reply nano/core_test/backlog.cpp:78]: Resorting to weak_ptr is a sign of not having a well managed lifetime of components and usually not necessary. It's possible to have that for IO threads too, the PR for that was mean to land after V24 release.

**Pull request closed without merging**